component/authenticator/src/main/java/org/wso2/carbon/identity/policy/password/PasswordChangeHandler.java

```diff
@@ -337,4 +359,50 @@ private String getClaimValue(UserStoreManager userStoreManager, String claimURI,
         return null;
     }
 
+    /**
+     * Check whether to store the last password update time in nanoseconds.
+     *
+     * @param tenantDomain
+     * @return true if the configuration is enabled
+     */
+    private boolean isStoreTimeAsNanoseconds(String tenantDomain) {
+
+        String storeTimeAsNanoseconds = PasswordPolicyUtils.getIdentityEventProperty(tenantDomain,
+                PasswordPolicyConstants.CONNECTOR_CONFIG_STORE_TIME_AS_NANOSECONDS);
+        if (storeTimeAsNanoseconds == null) {
+            return PasswordPolicyConstants.CONNECTOR_CONFIG_STORE_TIME_AS_NANOSECONDS_DEFAULT_VALUE;
+        }
+        return Boolean.parseBoolean(storeTimeAsNanoseconds);
+    }
+
+    /**
+     * Calculate the date difference between the current time and the password changed time.
+     *
+     * @param passwordChangedTime
+     * @param tenantDomain
+     * @return date difference
+     */
+    private int calculateDateDifference(long passwordChangedTime, String tenantDomain) {
+
+        // Check whether the password changed time is stored in nanoseconds.
+        boolean isNanoSeconds = passwordChangedTime >= 1_000_000_000_000_000L;
+
+        // Check whether the password changed time should be stored in nanoseconds or not.
+        if (isStoreTimeAsNanoseconds(tenantDomain)) {
+
+            // Convert the password changed time to nanoseconds if it is stored in milliseconds.
+            if (!isNanoSeconds) {
+                passwordChangedTime = passwordChangedTime * 1_000_000;
+            }
+            long currentTIme = System.currentTimeMillis() * 1000000;
```

Comment:
We may need to get the time using the Calendar like we have did previously because when the servers are running in different time zones, there can be inconsistencies. Please verify and update accordingly